android/app/src/main/java/org/fossasia/openevent/activities/auth/EditProfileActivity.java

```diff
@@ -112,72 +108,55 @@ private void saveChanges() {
     }
 
     private void uploadImage(String encodedImage) {
-        uploadImageDisposable = APIClient.getOpenEventAPI().uploadImage(new UploadImage(encodedImage))
-                .subscribeOn(Schedulers.io())
-                .observeOn(AndroidSchedulers.mainThread())
-                .subscribe(imageResponse -> {
-                            uploadedImageUrl = imageResponse.getUrl();
-                            Timber.d("Image uploaded successfully." + " Url: " + uploadImageDisposable);
-                        },
-                        throwable -> {
-                            showProgressBar(false);
-                            Toast.makeText(EditProfileActivity.this, R.string.error_uploading_image, Toast.LENGTH_SHORT).show();
-                            Timber.d("Error uploading image: " + throwable.getMessage());
-                        },
-                        () -> {
-                            showProgressBar(false);
-                            //Image uploaded successfully updating user...
-                            updateUser();
-                        },
-                        disposable -> showProgressBar(true));
+        showProgressBar(true);
+        editProfileActivityViewModel.uploadImage(encodedImage).observe(this, response ->{
+            switch (response){
+                case EditProfileActivityViewModel.IMAGE_UPLOAD_SUCCESSFUL:
+                    showProgressBar(false);
+                    updateUser();
+                    break;
+                case EditProfileActivityViewModel.IMAGE_UPLOAD_FAILED:
+                    showProgressBar(false);
+                    Toast.makeText(EditProfileActivity.this, R.string.error_uploading_image, Toast.LENGTH_SHORT).show();
+                    break;
+                case EditProfileActivityViewModel.IMAGE_UPLOAD_COMPLETE:
+                    showProgressBar(false);
```

Comment:
Move the showProgressBar(false) call outside of the switch statement as it is applicable to all three cases.

Reply:
yeah, that was silly! I will change that :D